### PR TITLE
feat: add api.tree.toggle_enable_filters

### PR DESCRIPTION
**Pull request by DeH4er**
Hi there. Thanks for the amazing project. 

I often find that I need to access hidden files but I keep forgetting keybindings for particular filters. I wish to have one goto keybinding to make it easier to see all files.

I'm unsure about the naming of some vars and functions. But overall, I hope it's a good contribution to the project

### CODE REVIEW

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Concept and change looks good.

Let's get some consistency in user and API naming. Perhaps `Toggle All Filters Disabled` and `api.tree.toggle_disable_all_filter()`

**Comment by DeH4er** [comment]
What do you think about`Toggle Filters Disabled` and `api.tree.toggle_filters_disabled`? PR is updated to reflect these names

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -536,6 +537,7 @@ Following is the default configuration. See |nvim-tree-opts| for details.
         show_on_open_dirs = true,
       },
       filters = {
+        disabled = false,
```
[the remark this answers is not in the document]
Comment (DeH4er) [reply doc/nvim-tree-lua.txt:541]: Actually, I see that the convention is `enable`. Will update PR

```
...
      hijack_directories = {
        enable = true,
        auto_open = true,
      },
      update_focused_file = {
        enable = false,
        update_root = false,
        ignore_list = {},
      },
      system_open = {
        cmd = "",
        args = {},
      },
      git = {
        enable = true,
        show_on_dirs = true,
        show_on_open_dirs = true,
        disable_for_dirs = {},
        timeout = 400,
        cygwin_support = false,
      },
...
```

File: doc/nvim-tree-lua.txt [hunk 2]
```diff
@@ -1242,6 +1244,11 @@ Only relevant when |modified.show_on_dirs| is `true`.
 
 File / folder filters that may be toggled.
 
+*nvim-tree.filters.enable*
+Enable / disable filters.
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:1248]:
```suggestion
Enable / disable all filters including live filter.
```

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Works nicely except for the live filter case.

I naturally tried that case when testing and was surprised when it didn't apply. Can we please add that? I imagine users will be similarly surprised.

**Comment by alex-courtis** [comment]
I'm not quite sure about the `,` mapping, I'm always nervous about overriding "core" vim keys.

@gegoune should we change it to something like `T`?

**Comment by gegoune** [comment]
> I'm not quite sure about the `,` mapping, I'm always nervous about overriding "core" vim keys.
> 
> @gegoune should we change it to something like `T`?

Not overriding existing mappings is a good thing for sure. I see two options:
1. map it to something free, I don;t really care what to be honest as there is a way to change the default
2. don't add new mappings going forward. We could even look into removing majority of existing mappings from `api.config.mappings.default_on_attach` providing only bare minimum. But that's radical, maybe too radical.

BTW, `T` is an existing vim key `:h T`.

**Comment by alex-courtis** [comment]
> 2. don't add new mappings going forward.

I'm inclined to agree. Each default mapping added is potentially a user's mapping clobbered.

Sorry @DeH4er - please remove the default mapping.

> We could even look into removing majority of existing mappings from api.config.mappings.default_on_attach providing only bare minimum. But that's radical, maybe too radical.

That would break people, however we could add a very trimmed down `api.config.mappings.minimal_on_attach`

**Comment by DeH4er** [comment]
I updated the PR with all your feedback.

Since you touched on a topic of keybindings, currently it's hard to get a sense of what's possible. 

1. The help panel is hidden under "g?" keybinding by default, and only when nvim-tree updated its way to change keybindings, I discovered that there's a help panel. 
2. As a user, you think everything useful (or even possible) is in this panel. 

So I think the ideal solution would be to:
1. Make it clear that g? is a default help panel keybinding. Show "g? - help" somewhere in nvim-tree panel or make it visible in readme.
2. Show every possible API command in this help panel, even if it has no keybinding + keybinding if there's one
3. Since there are lots of commands, it would be great to group them

This way, you can have minimal keybinding config and it will be extremely easy to copy-paste api function to your on_attach without reading :h nvim-tree

**Review by alex-courtis — APPROVED** [review]
Works beautifully, many thanks!

**Pull request merged**